### PR TITLE
fix: handle workspace name collisions

### LINKED ISSUE #590: bug: retry generated workspace names when branch or worktree already exists
## Platform

| Field | Value |
|---|---|
| **OS** | macOS from screenshot/report |
| **OS version** | Unknown |
| **Architecture** | Unknown |
| **Claudette version** | Unknown Discord report build |

## Description

Creating a new workspace can fail when the randomly generated workspace slug maps to a git branch or worktree path that already exists. The user clicked the New workspace button and got a native alert:

```
Failed to create workspace: Git command failed: Preparing worktree (new branch 'jasperm Mayone/dusty-dandelion')
fatal: a branch named 'jasperm Mayone/dusty-dandelion' already exists
```

The branch in the screenshot is `jaspermayone/dusty-dandelion`.

Root cause from code inspection:

- `src/ui/src/components/sidebar/Sidebar.tsx` and `src/ui/src/components/command-palette/CommandPalette.tsx` call `generateWorkspaceName()` once, then pass `generated.slug` directly to `createWorkspace(..., skipSetup=true)`.
- `src-tauri/src/commands/workspace.rs:create_workspace` builds `branch_name = format!("{prefix}{name}")` and `worktree_path = worktree_base / repo.path_slug / name` without checking existing branches, active/imported workspaces, or filesystem paths.
- `src/git.rs:create_worktree` always shells out to `git worktree add --no-track -b <branch_name> <worktree_path> <base>`. Git correctly fails when `refs/heads/<branch_name>` already exists, and the raw error bubbles back to the UI.
- The server remote path has the same shape in `src-server/src/handler.rs:handle_create_workspace`: it derives `branch_name = repo.path_slug/name` and calls the same git helper without uniqueness handling.

This is not the user exhausting all worktrees. It is a collision in our finite adjective/plant namespace, or a stale branch/worktree left behind from a previous workspace.

## Steps to Reproduce

1. Add/open a repository in Claudette.
2. Ensure the generated branch already exists, e.g. create `<configured-prefix>/dusty-dandelion` in the repo, or create/archive a workspace and leave/recreate the same branch condition.
3. Click the New workspace button until `dusty-dandelion` is generated, or force/mock the generator to return that slug.

## Expected Behavior

Claudette should not surface a raw git failure for an automatically generated name collision. It should generate another unused slug/branch/path and create the workspace, or present a clear recoverable error only after bounded retries are exhausted.

## Actual Behavior

Workspace creation fails with a native alert containing the git stderr:

```
fatal: a branch named '<prefix>/dusty-dandelion' already exists
```

## Logs / Screenshots

Discord report screenshot shows the native alert:

```
Failed to create workspace: Git command failed: Preparing worktree (new branch 'jaspermayone/dusty-dandelion')
fatal: a branch named 'jaspermayone/dusty-dandelion' already exists
```

Reporter notes:

- User clicked the button to make a new workspace.
- Existing workspace/branch may already have used that generated name.
- Expected behavior from discussion: catch this and generate a new workspace.

## Proposed Fix / Acceptance Criteria

- Centralize workspace slug allocation on the backend/Tauri side so every creation path uses the same uniqueness rule.
- Before calling `git worktree add -b`, reject candidates whose branch already exists, whose worktree path already exists, or whose active/imported workspace already uses the name/branch/path.
- Retry random name generation with a bounded limit, then fall back to a deterministic suffix such as `dusty-dandelion-2` if needed.
- Apply equivalent handling to remote/server workspace creation.
- Add tests for duplicate branch and duplicate path collisions.

### PR DESCRIPTION
**Pull request by jamesbrink**
## Summary

Fixes #590.

- Add shared backend workspace-name allocation with deterministic suffix fallback for occupied slugs.
- Check existing workspace names, branches, paths, git branches, registered worktrees, and on-disk paths before creating a worktree.
- Wire both local Tauri and remote server workspace creation through the shared allocator.
- Retry allocation when concurrent creation loses the race at `git worktree add`.
- Send `generated.slug` for remote workspace creation.

## Verification

- `cargo test --all-features`
- `cargo test -p claudette workspace_alloc::tests`
- `cargo test -p claudette git::tests::test_worktree_create_collision_classifier`
- `cargo fmt --all --check`
- `cargo clippy --workspace --all-targets`
- `cd src/ui && bunx tsc -b`
- `cd src/ui && bun run test`
- `cd src/ui && bun run lint:css`

Note: `cd src/ui && bun run lint` still fails on existing unrelated repo-wide React lint violations.

### CODE REVIEW

File: src-server/src/handler.rs [hunk 1]
```diff
@@ -788,39 +790,77 @@ async fn handle_create_workspace(
...
+                &branch_prefix,
+                worktree_base_dir.as_path(),
+            )
+            .await
+            .map_err(|e| e.to_string())?;
+
+            match claudette::git::create_worktree(
+                &repo.path,
+                &allocation.branch_name,
+                &allocation.worktree_path.to_string_lossy(),
```
[the remark this answers is not in the document]
Comment (jamesbrink) [reply src-server/src/handler.rs]: Fixed in 17d4778 by binding `allocation.worktree_path.to_string_lossy().to_string()` to an owned `worktree_path_str` before awaiting `create_worktree`. Re-ran focused allocator/classifier tests, fmt/diff checks, and clippy.
